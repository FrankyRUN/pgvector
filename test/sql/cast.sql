SELECT ARRAY[1,2,3]::vector;
SELECT ARRAY[1.0,2.0,3.0]::vector;
SELECT ARRAY[1,2,3]::float4[]::vector;
SELECT ARRAY[1,2,3]::float8[]::vector;
SELECT ARRAY[1,2,3]::numeric[]::vector;
SELECT '{NULL}'::real[]::vector;
SELECT '{NaN}'::real[]::vector;
SELECT '{Infinity}'::real[]::vector;
SELECT '{-Infinity}'::real[]::vector;
SELECT '{}'::real[]::vector;
SELECT '{{1}}'::real[]::vector;
SELECT '[1,2,3]'::vector::real[];
SELECT '[1,2,3]'::vector::halfvec;
SELECT '[1,2,3]'::halfvec::vector;
SELECT '[1,2,3]'::vector::halfvec(2);
SELECT '[1,2,3]'::halfvec::vector(2);
SELECT '[65520]'::vector::halfvec;
SELECT '[1e-8]'::vector::halfvec;
SELECT array_agg(n)::vector FROM generate_series(1, 16001) n;
SELECT array_to_vector(array_agg(n), 16001, false) FROM generate_series(1, 16001) n;

-- ensure no error
SELECT ARRAY[1,2,3] = ARRAY[1,2,3];
