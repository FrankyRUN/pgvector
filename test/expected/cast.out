SELECT ARRAY[1,2,3]::vector;
  array  
---------
 [1,2,3]
(1 row)

SELECT ARRAY[1.0,2.0,3.0]::vector;
  array  
---------
 [1,2,3]
(1 row)

SELECT ARRAY[1,2,3]::float4[]::vector;
  array  
---------
 [1,2,3]
(1 row)

SELECT ARRAY[1,2,3]::float8[]::vector;
  array  
---------
 [1,2,3]
(1 row)

SELECT ARRAY[1,2,3]::numeric[]::vector;
  array  
---------
 [1,2,3]
(1 row)

SELECT '{NULL}'::real[]::vector;
ERROR:  array must not contain nulls
SELECT '{NaN}'::real[]::vector;
ERROR:  NaN not allowed in vector
SELECT '{Infinity}'::real[]::vector;
ERROR:  infinite value not allowed in vector
SELECT '{-Infinity}'::real[]::vector;
ERROR:  infinite value not allowed in vector
SELECT '{}'::real[]::vector;
ERROR:  vector must have at least 1 dimension
SELECT '{{1}}'::real[]::vector;
ERROR:  array must be 1-D
SELECT '[1,2,3]'::vector::real[];
 float4  
---------
 {1,2,3}
(1 row)

SELECT '[1,2,3]'::vector::halfvec;
 halfvec 
---------
 [1,2,3]
(1 row)

SELECT '[1,2,3]'::halfvec::vector;
 vector  
---------
 [1,2,3]
(1 row)

SELECT '[1,2,3]'::vector::halfvec(2);
ERROR:  expected 2 dimensions, not 3
SELECT '[1,2,3]'::halfvec::vector(2);
ERROR:  expected 2 dimensions, not 3
SELECT '[65520]'::vector::halfvec;
ERROR:  infinite value not allowed in halfvec
SELECT '[1e-8]'::vector::halfvec;
 halfvec 
---------
 [0]
(1 row)

SELECT array_agg(n)::vector FROM generate_series(1, 16001) n;
ERROR:  vector cannot have more than 16000 dimensions
SELECT array_to_vector(array_agg(n), 16001, false) FROM generate_series(1, 16001) n;
ERROR:  vector cannot have more than 16000 dimensions
-- ensure no error
SELECT ARRAY[1,2,3] = ARRAY[1,2,3];
 ?column? 
----------
 t
(1 row)

